### ISSUE TITLE
Regression in Swagger Codegen 3 vs 2: Nested properties have type 'Object' and the corresponding model is not generated

### ISSUE THREAD

**Issue by janeisklar** [swagger-api/swagger-codegen-generators#423]
Given the following swagger file with nested objects:
```yaml
---
swagger: "2.0"
info:
  description: "Foo"
  version: "1.0.0"
host: "something.com"
basePath: "/"
schemes:
- "https"
consumes:
- "application/json"
produces:
- "application/json"
paths:
  /example:
    get:
      responses:
        200:
          description: "OK"
          schema:
            $ref: "#/definitions/Foo"
    parameters: []
definitions:
  Foo:
    type: "object"
    required:
    properties:
      nested:
        type: "object"
        properties:
          color:
            type: "string"
```

Swagger 2.0 would generate a java class like this:
```java
public class Foo   {
  @JsonProperty("nested")
  private FooNested nested = null;
  ...
}
```

With Swagger 3.0, however, something like this will be generated:
```java
public class Foo   {
  @JsonProperty("nested")
  private Object nested = null;
  ...
}
```

Consequently, serialisation and deserialisation of a nested property will not work with Swagger 3. Furthermore, the class `FooNested` will not be generated anymore either.

For reference here are the plugin specifications I've used for the Swagger 2 example:
```xml
<plugin>
    <groupId>io.swagger</groupId>
    <artifactId>swagger-codegen-maven-plugin</artifactId>
    <version>2.4.6</version>
    <executions>
        <execution>
            <goals>
                <goal>generate</goal>
            </goals>
            <configuration>
                <inputSpec>${project.basedir}/src/main/resources/example.yaml</inputSpec>
                <language>spring</language>
                <library>spring-boot</library>
                <modelPackage>old.swagger.model</modelPackage>
                <apiPackage>old.swagger.api</apiPackage>
                <generateApiDocumentation>false</generateApiDocumentation>
                <generateSupportingFiles>false</generateSupportingFiles>
                <configOptions>
                    <sourceFolder>src/gen/java/main</sourceFolder>
                    <dateLibrary>java8</dateLibrary>
                </configOptions>
            </configuration>
        </execution>
    </executions>
</plugin>
```

And for Swagger 3:
```xml
<plugin>
    <groupId>io.swagger.codegen.v3</groupId>
    <artifactId>swagger-codegen-maven-plugin</artifactId>
    <version>3.0.8</version>
    <executions>
        <execution>
            <goals>
                <goal>generate</goal>
            </goals>
            <configuration>
                <inputSpec>${project.basedir}/src/main/resources/example.yaml</inputSpec>
                <language>spring</language>
                <library>spring-boot</library>
                <modelPackage>new.swagger.model</modelPackage>
                <apiPackage>new.swagger.api</apiPackage>
                <generateApiDocumentation>false</generateApiDocumentation>
                <generateSupportingFiles>false</generateSupportingFiles>
                <configOptions>
                    <sourceFolder>src/gen2/java/main</sourceFolder>
                    <dateLibrary>java8</dateLibrary>
                </configOptions>
            </configuration>
        </execution>
    </executions>
</plugin>
```

**Comment by shexbeer** [comment]
Any news here? This is a blocker from using the new generators

**Comment by frantuma** [comment]
thanks for reporting this. It happens when processing a Swagger/OpenAPI 2.0 spec using swagger-codegen 3.0.0 generators; in this scenario the spec is first converted to 3.0.0 (by [swagger-parser](https://github.com/swagger-api/swagger-parser)) and then used as input to 3.0.0 generators; 

this has been fixed in swagger-api/swagger-parser#1181 available in latest snapshot and upcoming release.

Closing ticket, please reopen if you're still experiencing issues
